### ISSUE TITLE
dannce-predict loss very small, but result same like normal but shift

### ISSUE THREAD

**Issue by yuan0821** [spoonsso/dannce#138]
hi @davidhildebrand @spoonsso @data-hound 

The loss of the prediction result is very low, and the relative position of each label is correct, but the result seems to be shifted as a whole, which does not match the video image. But the result of com-predict looks normal.
Thank you so much for your time...
`
viewGui = View3D(params, videos, skeleton);

pts3d = load('F:\dannce\demo\mouse20230129\DANNCE\predict_results\save_data_MAX0.mat')

viewGui.loadFrom3D(pts3d.pred) 
`

<img width="1467" alt="截屏2023-02-08 上午1 12 23" src="https://user-images.githubusercontent.com/50692530/217318472-04e55bc1-0789-4a02-a367-40765b162316.png">

<img width="1469" alt="截屏2023-02-08 上午1 31 39" src="https://user-images.githubusercontent.com/50692530/217320431-621e9fe7-3051-49f6-9e58-68b11c2160d7.png">

**Comment by spoonsso** [comment]
Can you plot the projected COM onto these images as well? Are you using a
MAX model or an AVG model? If a MAX model, you may need to add the 3D COM
coordinates to the predictions, depending on which version/branch you are
using.

**Comment by yuan0821** [comment]
Thank you @spoonsso 
I start with MAX and finetune with a 5-cam-pre-trained MAX network.
I have four cameras, so I duplicate the camera1 video file as camera5. And also, copy the sync and label data in the .mat
```
net_type: MAX
train_mode: finetune
num_validation_per_exp: 0
max_num_samples: 1000
dannce_finetune_weights: 'F:\dannce\weight\MAX5\'
```
Please advise how to load com and dance predict results in the same image. 
I use the dannce_development branch.
[https://github.com/spoonsso/dannce/tree/release_development]

And I also don't know how to add the 3d COM coordinates to the prediction; only use:
`new_predict=pred(:, :,1:16)+com(:,:);`???
After the addition,
the result same a little better, but still shifted.
<img width="1466" alt="截屏2023-02-08 上午2 04 56" src="https://user-images.githubusercontent.com/50692530/217329049-6cac8417-793d-4142-b990-ed461ffa084c.png">

**Comment by yuan0821** [comment]
<img width="1326" alt="截屏2023-02-08 上午2 27 38" src="https://user-images.githubusercontent.com/50692530/217333547-bbf04db8-71a9-4a05-8682-2215756b1feb.png">
both in the frame 4 between com and dannce predict result.

**Comment by yuan0821** [comment]
hi @spoonsso 
thank you for suggestion. the result is normal now after add COM corrdinates
```
for i=1:16
    new_predict(:,:,i)=dannce_predict.pred(:,:,i)+com_predict.com;
end

skeleton = load('F:\dannce\Label3D\skeletons\rat16.mat');
close all
viewGui = View3D(params, videos, skeleton);
viewGui.loadFrom3D(new_predict(framesToLabel,:,:));
```
<img width="1453" alt="截屏2023-02-08 上午2 39 49" src="https://user-images.githubusercontent.com/50692530/217336016-079fee5f-b9c0-48f9-b933-c2c4bae43d6d.png">

**Comment by spoonsso** [comment]
Perfect, glad it is working now. By the way, the reason why the end of your tail is not being tracked well is because it is most likely outside of the 3D extent of the image volume.

@data-hound there is an `addCOM` argument in `processing.savedata_tomat()` that would solve this problem during dannce prediction, but it looks like it is not being used. Can we add this to the release you are working on?

**Comment by data-hound** [comment]
Yes, we can add this `addCOM` argument to the `savedata_tomat()`. This should be done for MAX model predictions, correct?

**Comment by spoonsso** [comment]
Yes, the AVG predictions already have the COM built in b/c they use the
world grid coordinates in the calculation.
